hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/PlanExecutor.java

```diff
@@ -157,28 +171,29 @@ SqlResult execute(DropMappingPlan plan) {
     }
 
     SqlResult execute(CreateDataLinkPlan plan) {
-        InternalDataLinkService dlService = getNodeEngine(hazelcastInstance).getDataLinkService();
-        if (plan.ifNotExists() && dlService.existsDataLink(plan.name())) {
-            throw new HazelcastException("Data link '" + plan.name() + "' already exists");
-        }
+        InternalDataLinkService dlService = nodeEngine.getDataLinkService();
+        assert !plan.ifNotExists() || !plan.isReplace();
 
-        dlService.createSqlDataLink(plan.name(), plan.type(), plan.options(), plan.isReplace());
-        dataLinksCatalog.createDataLink(
-                new DataLink(plan.name(), plan.type(), plan.options()),
+        if (dlService.existsConfigDataLink(plan.name())) {
+            throw new HazelcastException("Cannot replace a data link created from configuration");
```

Comment:
Do we care for cases when nodes have different static data link configurations? Probably not.

Reply:
I think there's some check that the configurations on nodes are equal when a node is joining the cluster.